packages/lib-user/src/components/UserHome/components/Dashboard/DashboardContainer.js

```diff
@@ -0,0 +1,40 @@
+import { panoptes } from '@zooniverse/panoptes-js'
+import auth from 'panoptes-client/lib/auth'
+import useSWR from 'swr'
+
+import Dashboard from './Dashboard'
+
+const SWROptions = {
+  revalidateIfStale: true,
+  revalidateOnMount: true,
+  revalidateOnFocus: true,
+  revalidateOnReconnect: true,
+  refreshInterval: 0
+}
+
+/* This is a similar pattern to usePanoptesUser hook, but includes the profile_header */
+const fetchProfileBanner = async ({ authUser }) => {
+  const token = await auth.checkBearerToken()
+  const authorization = `Bearer ${token}`
+  const query = { include: 'profile_header', user_id: authUser.id }
+
+  try {
+    const { body } = await panoptes.get('/users', query, { authorization })
+    const user = body.users?.[0]
+
+    if (body.linked?.profile_headers?.length) {
+      user.profile_header = body.linked.profile_headers[0].src
+    }
+    return user
+  } catch (error) {
+    console.error(error)
+    return null
+  }
+}
+
+export default function DashboardContainer({ authUser }) {
+  const key = { authUser }
```

Comment:
If `authUser` is empty or null, you should set the key to null. That tells SWR to skip the request.

Otherwise, your code here might inadvertently run the SWR hook when it shouldn't, and cache an unexpected value for this key in the SWR cache. 

When you pass a key to `useSWR`, it looks up that key in its global cache and immediately returns the stored result (which is why keys must be unique.) If the stored result is stale, it revalidates the cache by fetching a new result from the remote source. Hence 'stale while revalidate.'

Reply:
Also a warning that if you've used `{ authUser }` as the key for any other queries, elsewhere in your code, `useSWR(key, fetcher, options)` will returned the cached values for those queries. That can lead to some weird bugs.

Reply:
DashboardContainer is in UserHome, and UserHome only renders if `authUser.login` exists. I don't expect `authUser` to ever be null in this component.

UserHome is conditionally rendered in app-root here. Do you suggest handling the homepage differently?
https://github.com/zooniverse/front-end-monorepo/blob/4deea7e0476c7352e768da6eaadfed1ee0ccf169/packages/app-root/src/components/HomePageContainer.js#L24

Reply:
Sounds good. Good catch with the need for a unique `key` too, so I'll definitely address that here!

Reply:
I think that unique key caught me out here, which is why I added the endpoint.
https://github.com/zooniverse/front-end-monorepo/blob/4deea7e0476c7352e768da6eaadfed1ee0ccf169/packages/lib-react-components/src/hooks/usePanoptesUser.js#L32-L35